### PR TITLE
Improvements to file list UI

### PR DESCRIPTION
**Pull request by rbren**
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**

Improvements to file list UI

* Don't show the agent as loaded until the runtime is ready
  * Currently the UI shows an empty workspace even when it says it's ready
* Filter files based on gitignore at the API level, not the runtime level
  * Runtime should have access to *all* files
* Always filter .git
* Show a "loading" message instead of zero state when waiting for files
---

**Give a summary of what the PR does, explaining any non-trivial design decisions**

Most controversial thing here is probably running `wait_until_alive` in the runtime constructor.

---
**Link of any specific issues this addresses**

### CODE REVIEW

**Comment by rbren** [comment]
@xingyaoww would love your thoughts on whether:
* it's OK to move the file-hiding logic out of the runtime (IMO this is important--agent might want to e.g. look inside node_modules to debug)
* it's OK to wait for the runtime to be ready inside the constructor

**Comment by tobitege** [comment]
@rbren maybe we could put some effort together? :) 
https://github.com/All-Hands-AI/OpenHands/pull/3771

File: openhands/server/listen.py [hunk 1]
```diff
@@ -407,6 +414,23 @@ async def list_files(request: Request, path: str | None = None):
...
+            spec = PathSpec.from_lines(
+                GitWildMatchPattern, observation.content.splitlines()
+            )
+        except Exception as e:
+            print(e)
+            return file_list
+        file_list = [entry for entry in file_list if not spec.match_file(entry)]
+        return file_list
+
+    file_list = filter_for_gitignore(file_list, '')
```
Comment (tobitege) [inline openhands/server/listen.py:433]: The check for the .gitignore actually existing is missing completely. Also, the fallback is gone. Why?

Comment (rbren) [reply openhands/server/listen.py:433]: The check is now in `try/catch`--if it doesn't exist, we just bail out. I could probably be more explicit there though

The fallback is an antipattern IMO. Hiding a random list of folders that the user might expect to be available is potentially very confusing, and there was a lot in there

The only folder we should hide by default is `.git`

Comment (rbren) [reply openhands/server/listen.py:433]: Ah interesting. Do you have any record here? Would love to take a look

Hard for me to imagine folks not just adding those to .gitignore...

Comment (tobitege) [reply openhands/server/listen.py:433]: The only PR I could find on short notice is #2621

Comment (rbren) [reply openhands/server/listen.py:433]: Adding things like `.DS_Store` does make sense to me. But `logs`, `dist`, etc seem too aggressive--I have projects where I check `dist` into the repo for example, and I'd be very confused why it's not there. So I'm curious what folders folks specifically asked to be hidden

Comment (tobitege) [reply openhands/server/listen.py:433]: I didn't mean that each entry in the fallback list was specifically requested to be ignored, just that the "clutter" in the UI should be less. I think back then I first added that hard-coded list and in a follow up PR the .gitignore support, but kept that list as the fallback. Most of those entries were based on _my_ user perspective, so I understand if this list may not be representative for all scenarios, they should be removed.

Comment (rbren) [reply openhands/server/listen.py:433]: Ahh got it! Thanks for the context. I'll add back in a few of the most obvious defaults. But I think we should keep it very minimal--much better to have clutter than confusion :)

File: openhands/runtime/client/client.py [hunk 2]
```diff
@@ -649,52 +647,12 @@ async def list_files(request: Request):
             if not os.path.exists(full_path) or not os.path.isdir(full_path):
                 return []
 
-            # Check if .gitignore exists
-            gitignore_path = os.path.join(full_path, '.gitignore')
```
Comment (tobitege) [inline openhands/runtime/client/client.py:653]: I still think, at least the gitignore check should not be removed for the UI here.

Comment (tobitege) [reply openhands/runtime/client/client.py:653]: The client is running inside the container and still has access to the mounted workspace folder, no?
Or are you thinking of the `remote` runtime? 🤔

Comment (tobitege) [reply openhands/runtime/client/client.py:653]: Thanks, that indeed makes it clearer. :)

**Review by tobitege — APPROVED** [review]
LGTM

**Pull request merged**